client/src/components/Panel.tsx

```diff
@@ -14,7 +14,59 @@ import "./Panel.scss";
 
 const { text_maker, TM } = create_text_maker_component(text);
 
-const PanelSource = ({ links }) => {
+interface Link {
+  href: string;
+  html: string;
+}
+interface PanelSourceProps {
+  links: Link[];
+}
+
+interface PanelGlossaryProps {
+  keys: string[];
+}
+
+interface Constructor {
+  plural: string;
+  singular: string;
+  subject_type: string;
+}
+interface Subject {
+  constructor: Constructor;
+  description: string;
+  guid: string;
+  has_planned_spending: boolean;
+  id: string;
+  is: (comparator: any) => boolean;
+  legal_title: string;
+  level: string;
+  lookup: Function;
+  name: string;
+  plural: string;
+  singular: string;
+  subject_type: string;
+}
+interface Footnote {
```

Comment:
I would rather you drop converting Panel.js for now for the same reason as ensure_loaded.
These footnotes, subject should be built from the ground up